### DISCUSSION TITLE
Rerun failed jobs doesn't publish check suites

### DISCUSSION THREAD

**Discussion by TWiStErRob** [community/community#13480] [Actions] [answered]
https://github.com/TWiStErRob/net.twisterrob.colorfilters/actions/runs/2037116362 is a good example.

This execution was on a pushed commit of a PR branch to default branch, here are the branch protection rules:
![image](https://user-images.githubusercontent.com/2906988/160084674-bcabf79e-358b-4470-b424-f4f83a38f382.png)

Workflow file: https://github.com/TWiStErRob/net.twisterrob.colorfilters/blob/503ca81dd18064b75990489d3f6298c757d452ac/.github/workflows/CI.yml

Workflow structure is quite simple: two jobs resulting in one check each + a code scan:
🔨 Build & Verify &rarr; 🔔 Test: Unit Results
🧪 Instrumentation Tests &rarr; 🔔 Test: Instrumentation Results
🔨 Build & Verify &rarr; "detekt" code scanning

[Attempt #1](https://github.com/TWiStErRob/net.twisterrob.colorfilters/actions/runs/2037116362/attempts/1): Build & Verify timed out so there's no Unit Results.
![image](https://user-images.githubusercontent.com/2906988/160082972-f173ae17-1cc6-483b-a431-84ebae84d818.png)

[Attempt #2](https://github.com/TWiStErRob/net.twisterrob.colorfilters/actions/runs/2037116362/attempts/2): Re-ran failed jobs (i.e. Build & Verify), which succeeded and produced Unit Results as expected, but the Instrumentation job's check Results were not propagated.
![image](https://user-images.githubusercontent.com/2906988/160083016-25909ec5-5998-48d1-811a-3afea2b47fd3.png)

[Attempt #3](https://github.com/TWiStErRob/net.twisterrob.colorfilters/actions/runs/2037116362/attempts/3): Re-ran all jobs to get both results.
![image](https://user-images.githubusercontent.com/2906988/160083066-6b76a9d5-bb57-4305-bad7-59f8da0c18b0.png)

At Attempt 2 this is how the PR looked like:
![image](https://user-images.githubusercontent.com/2906988/160083087-ea84051b-61f7-4b38-8a73-44bbad4c1d83.png)
This means that I wasn't able to merge even though all jobs and checks have succeeded in the workflow, because the Instrumentation job's check Results were not propagated to later attempts.

(I just noticed while reporting that detekt was also missing the same way.)

**Comment by chrispat** [comment, marked as answer]
Than you for the feedback.

I have filed an issue and we are looking into it.

**Reply by mcarrillowz** [reply]
Hi 👋 , 

Is there any update on this?
